CPAProxyTests/CPAProxyManagerTests.m

```diff
@@ -190,6 +202,14 @@ - (void) socket:(GCDAsyncSocket *)sock didReadData:(NSData *)data withTag:(long)
 
 - (void) socketDidDisconnect:(GCDAsyncSocket *)sock withError:(NSError *)err {
     NSLog(@"socketDidDisconnect %@", err);
+    if (err.code == 4) { // Sometimes the socket connects but cannot read data and keeps failing, I suspect the IP is rejected by the server
+        NSError *error = nil;
+        [((GCDAsyncProxySocket *) sock) initWithDelegate:self delegateQueue:dispatch_get_main_queue()];
```

Comment:
Yes, but I found that if I didn't do that and just tried to reconnect there, it would keep trying and failing until the `waitForExpectationsWithTimeout` timed out. I couldn't create a new socket there since it would become nil as soon as it exited there.

Is an approach with socket variables that are more persistent more desirable?